core/store/orm/orm.go

```diff
@@ -460,10 +461,40 @@ WHERE run_results.id = updates.id
 
 // CreateJobRun inserts a new JobRun
 func (orm *ORM) CreateJobRun(run *models.JobRun) error {
-	if err := orm.MustEnsureAdvisoryLock(); err != nil {
-		return err
-	}
-	return orm.DB.Create(run).Error
+	return orm.convenientTransaction(func(tx *gorm.DB) error {
+		err := tx.Clauses(clause.OnConflict{DoNothing: true}).Create(&run.RunRequest).Error
```

Comment:
Note that on conflict this will not populate the ID of the passed run from the database, or load any data for that matter. Is this ok?

Reply:
Without auto associations that is now true for _every association_ so it is now done manually as needed everywhere.